scipy/sparse/construct.py

```diff
@@ -405,18 +405,20 @@ def _compressed_sparse_stack(blocks, axis):
     other_axis = 1 if axis == 0 else 0
     data = np.concatenate([b.data for b in blocks])
     indices = np.concatenate([b.indices for b in blocks])
-    indptr = []
-    last_indptr = 0
+    idx_dtype = get_index_dtype(arrays=[b.indptr for b in blocks],
+                                maxval=data.size)
+    indptr = np.empty(np.sum([b.shape[axis] for b in blocks])+1,
+                      dtype=idx_dtype)
+    last_indptr = idx_dtype(0)
```

Comment:
This can be `last_indptr = 0` --- using a numpy dtype scalar here is not necessary, as the type of the scalar does not matter in additions.